File: 2 Zadanie/app/src/main/java/com/example/a2zadanie/MainActivity.java
package com.example.a2zadanie;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    public static final String KEY_EXTRA_ANSWER = "com.example.a2zadanie.CorrectAnswer";
    private static final String KEY_CURRENT_POINTS = "points";
    private static final String KEY_CURRENT_INDEX = "currentIndex";
    private static final int REQUEST_CODE_PROMPT = 0;
    private Button trueButton;
    private Button falseButton;
    private Button nextButton;
    private Button promptButton;
    private TextView questionTextView;
    private Question[] questions = new Question[]{
            new Question(R.string.q_activity, true),
            new Question(R.string.q_find_resources, false),
            new Question(R.string.q_listener, true),
            new Question(R.string.q_resources, true),
            new Question(R.string.q_version, false)
    };
    private int currentIndex = 0;
    private int points = 0;
    private boolean answerWasShown;
    private void checkAnswerCorrectness(boolean userAnswer){
        boolean correctAnswer = questions[currentIndex].isTrueAnswer();
        int resultMessageId = 0;
        if(answerWasShown){
            resultMessageId = R.string.answer_was_shown;
        }else{
            if(userAnswer == correctAnswer){
                resultMessageId = R.string.correct_answer;
                points++;
            }else{
                resultMessageId = R.string.incorrect_answer;
            }
        }
        Toast.makeText(this,resultMessageId,Toast.LENGTH_SHORT).show();
    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d("call","onCreate");
        setContentView(R.layout.activity_main);
        if (savedInstanceState !=null)
        {
            currentIndex = savedInstanceState.getInt(KEY_CURRENT_INDEX);
        }
        trueButton = findViewById(R.id.true_button);
        falseButton = findViewById(R.id.false_button);
        nextButton = findViewById(R.id.next_button);
        questionTextView = findViewById(R.id.question_text_view);
        promptButton = findViewById(R.id.answer_Button);

        promptButton.setOnClickListener((v) -> {
            Intent intent = new Intent(MainActivity.this, PromptActivity.class);
            boolean correctAnswer = questions[currentIndex].isTrueAnswer();
            intent.putExtra(KEY_EXTRA_ANSWER, correctAnswer);
            startActivityForResult(intent, REQUEST_CODE_PROMPT);
        });
        trueButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                checkAnswerCorrectness(true);
            }
        });
        falseButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                checkAnswerCorrectness(false);
            }
        });
        nextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                answerWasShown = false;
                if(currentIndex==4)
                {
                    questionTextView.setText("Zdobyłeś "+points+" pkt");
                }
                else{
                    currentIndex = (currentIndex+1)%questions.length;
                    setNextQuestion();
                }

            }
        });
        setNextQuestion();
    }
    private void setNextQuestion(){
        questionTextView.setText(questions[currentIndex].getQuestionId());
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d("call", "onStart");
    }
    @Override
    protected void onPause() {
        super.onPause();
        Log.d("call","onPause");
    }
    @Override
    protected void onResume() {
        super.onResume();
        Log.d("call","OnResume");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d("call","OnStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d("call","OnDestroy");
    }

    @Override
    protected void onSaveInstanceState(Bundle outState){
        super.onSaveInstanceState(outState);
        Log.d("call","Wywołana została metoda: onSaveInstanceState");
        outState.putInt(KEY_CURRENT_INDEX, currentIndex);
        outState.putInt(KEY_CURRENT_POINTS, points);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if(resultCode != RESULT_OK) {
            return ;
        }
        if(requestCode == REQUEST_CODE_PROMPT){
            if(data == null) {
                return;
            }
            answerWasShown = data.getBooleanExtra(PromptActivity.KEY_EXTRA_ANSWER_SHOWN, false);
        }
    }

}

File: 1 Zadanie/App/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.myapplication.ui.theme.MyApplicationTheme


class MainActivity : ComponentActivity() {
    private lateinit var trueButton: Button
    private lateinit var falseButton: Button
    private lateinit var nextButton: Button
    private lateinit var questionTextView: TextView


    private val questions = listOf(
        Question(R.string.q_activity, true),
        Question(R.string.q_find_resources, false),
        Question(R.string.q_listener, true),
        Question(R.string.q_resources, true),
        Question(R.string.q_version, false)
    )

    private var currentIndex = 0

    private fun checkAnswerCorrectness(userAnswer: Boolean) {
        val correctAnswer = questions[currentIndex].isTrueAnswer()
        val resultMessageId: Int = if (userAnswer == correctAnswer) {
            R.string.correct_answer
        } else {
            R.string.incorrect_answer
        }
        Toast.makeText(this,resultMessageId,Toast.LENGTH_SHORT).show()
    }

    private fun setNextQuestion() {
        currentIndex = (currentIndex + 1) % questions.size
        displayQuestion()
    }

    private fun displayQuestion() {
        questionTextView.setText(questions[currentIndex].getQuestionId())
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

                    trueButton = findViewById(R.id.true_button)
                    falseButton = findViewById(R.id.false_button)
                    nextButton = findViewById(R.id.next_button)
                    questionTextView = findViewById(R.id.question_text_view)

                    trueButton.setOnClickListener {
                        checkAnswerCorrectness(true)
                    }

                    falseButton.setOnClickListener {
                        checkAnswerCorrectness(false)
                    }

                    nextButton.setOnClickListener {
                        currentIndex = (currentIndex + 1) % questions.size
                        setNextQuestion()
                    }

                    setNextQuestion()

    }
}

